src/optimizer.rs

```diff
@@ -99,27 +95,25 @@ pub fn optimize(
     let mut index = 0;
     let mut old_index = 0;
 
-    {
-        loop {
-            if type_section(module)
-                .map(|section| section.types_mut().len())
-                .unwrap_or(0)
-                == index
-            {
-                break;
-            }
+    loop {
+        if type_section(module)
+            .map(|section| section.types_mut().len())
+            .unwrap_or(0)
+            == index
+        {
+            break;
+        }
 
-            if stay.contains(&Symbol::Type(old_index)) {
-                index += 1;
-            } else {
-                type_section(module)
+        if stay.contains(&Symbol::Type(old_index)) {
+            index += 1;
+        } else {
+            type_section(module)
 					.expect("If type section does not exists, the loop will break at the beginning of first iteration")
 					.types_mut().remove(index);
-                eliminated_types.push(old_index);
-                trace!("Eliminated type({})", old_index);
-            }
-            old_index += 1;
+            eliminated_types.push(old_index);
+            trace!("Eliminated type({})", old_index);
```

Comment:
The `else` branch could use some formatting, maybe use the `cargo fmt` on the entire code.

Reply:
Yeah, original authors used tabs, but we tried to normalize it to indent with spaces.